File: Books.java
public class Books {
    private String  id;
    private String title;
    private String author;
    private int year;
    private boolean itsAvailable = true;

    public Books(String id, String title, String author, int year) {
        this.id = id;
        this.title = title;
        this.author = author;
        this.year = year;
    }

    @Override
    public String toString() {
        return "Books{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", author='" + author + '\'' +
                ", year=" + year +
                '}';
    }
    public static void registerBook(){
        ConsoleReader.sc.nextLine();
        System.out.println("Title : ");
        String title = ConsoleReader.sc.nextLine();
        System.out.println("Author : ");
        String author = ConsoleReader.sc.nextLine();
        if (title.isEmpty()&&author.isEmpty()){
            System.out.println("Invalid input, please introduce correctly the data");
            return;
        }
        System.out.println("Publication year : ");
        int year = ConsoleReader.sc.nextInt();
        if(year>2024){
            System.out.println("Invalid input, please verify");
            return;
        }
        String id= IdGenerator.createBookId();
        Books newBook = new Books(id, title, author, year );
        Library.addBook(newBook);
        System.out.println("Book registered succesfully");
    }

    public int getYear() {
        return year;
    }

    public String getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }
    public String getAuthor() {
        return author;
    }

    public boolean isItsAvailable() {
        return itsAvailable;
    }

    public void setItsAvailable(boolean itsAvailable) {
        this.itsAvailable = itsAvailable;
    }

}

File: books.py

from code_generator import CodeGenerator


class Books:

    def __init__(self, identifier, title, author, year):
        self.id = identifier
        self.title = title
        self.author = author
        self.year = year
        self.its_available = True
    
    def get_year(self):
        return self.year

    def get_id(self):
        return self.id

    def get_title(self):
        return self.title

    def get_author(self):
        return self.author

    def is_its_available(self):
        return self.its_available

    def set_its_available(self, its_available):
        self.its_available = its_available
